File: Constructorio_Net/src/models/ResultData.java
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

/**
 * Constructor.io Result Data
 **/
namespace Constructorio_NET
{
    public class ResultData {

        [JsonPropertyName("description")]
        public String description;

        [JsonPropertyName("id")]
        public String id;

        [JsonPropertyName("url")]
        public String url;

        [JsonPropertyName("image_url")]
        public String imageUrl;

        [JsonPropertyName("groups")]
        public List<ResultGroup> groups;

        [JsonPropertyName("facets")]
        public List<ResultFacet> facets;

        [JsonPropertyName("variation_id")]
        public String variationId;

        [JsonPropertyName("metadata")]
        public Map<String, Object> metadata;

    }
}

File: src/Constructorio_NET/models/ResultData.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;

/**
 * Constructor.io Result Data
 */
namespace Constructorio_NET.Models
{
    public class ResultData
    {
        [JsonProperty("description")]
        public string Description { get; set; }

        [JsonProperty("id")]
        public string Id { get; set; }

        [JsonProperty("url")]
        public string Url { get; set; }

        [JsonProperty("image_url")]
        public string ImageUrl { get; set; }

        [JsonProperty("groups")]
        public List<ResultGroup> Groups { get; set; }

        [JsonProperty("facets")]
        public List<ResultFacet> Facets { get; set; }

        [JsonProperty("variation_id")]
        public string VariationId { get; set; }

        [JsonExtensionData]
        public Dictionary<string, object> Metadata { get; set; }
    }
}